### ISSUE TITLE
[package] libiconv/1.17: cannon recognize windows subsystem

### ISSUE THREAD

**Issue by TheClonerx** [conan-io/conan-center-index#12393]
### Package and Environment Details

* Package Name/Version: **libiconv/1.17**
* Operating System+version: **Windows 10**
* Compiler+version: **GCC 12.1 (x86_64-win32-seh-rev3, Built by MinGW-W64 project)**
* Docker image: **N/A**
* Conan version: **conan 1.51.3**
* Python version: **Python 3.10.5**

### Conan profile

```
[settings]
os=Window
arch=x86_64
compiler=gcc
compiler.version=12.1
compiler.libcxx=libstdc++11
compiler.threads=win32
compiler.exception=seh
build_type=Release
[conf]
tools.microsoft.bash:path=/usr/bin/bash
[env]
CONAN_BASH_PATH=/usr/bin/bash
```

### Steps to reproduce

Install MinGW from <https://github.com/niXman/mingw-builds-binaries/releases>
Install CMake from <https://cmake.org/download/>
Install Git from <https://git-scm.com/downloads>

Open a git bash terminal an run `conan install 1.17 -u -b missing -pr:b default -pr:h default`

### Logs

<details><summary>Click to expand log</summary>
```
Configuration (profile_host):
[settings]
arch=x86_64
build_type=Release
compiler=gcc
compiler.exception=seh
compiler.libcxx=libstdc++11
compiler.threads=win32
compiler.version=12.1
os=Windows
[options]
[build_requires]
[env]
CONAN_BASH_PATH=/usr/bin/bash
[conf]
tools.microsoft.bash:path=/usr/bin/bash

Configuration (profile_build):
[settings]
arch=x86_64
build_type=Release
compiler=gcc
compiler.exception=seh
compiler.libcxx=libstdc++11
compiler.threads=win32
compiler.version=12.1
os=Windows
[options]
[build_requires]
[env]
CONAN_BASH_PATH=/usr/bin/bash
[conf]
tools.microsoft.bash:path=/usr/bin/bash

libiconv/1.17: WARN: Package binary is corrupted, removing: 11f724de2fd493a602fe99bef392fdba32b06d66
Installing package: libiconv/1.17
Requirements
    libiconv/1.17 from 'conancenter' - Cache
Packages
    libiconv/1.17:11f724de2fd493a602fe99bef392fdba32b06d66 - Build

Installing (downloading, building) binaries...
libiconv/1.17: WARN: Build folder is dirty, removing it: C:\Users\me\.conan\data\libiconv\1.17\_\_\build\11f724de2fd493a602fe99bef392fdba32b06d66
libiconv/1.17: Copying sources to build folder
libiconv/1.17: Building your package in C:\Users\me\.conan\data\libiconv\1.17\_\_\build\11f724de2fd493a602fe99bef392fdba32b06d66
libiconv/1.17: Generator txt created conanbuildinfo.txt
libiconv/1.17: Aggregating env generators
libiconv/1.17: Calling build()
libiconv/1.17: WARN: Error running `configure --help`: Cannot recognize the Windows subsystem, install MSYS2/cygwin or specify a build_require to apply it.
libiconv/1.17: Calling:
 > /c/users/me/.conan/data/libiconv/1.17/_/_/build/11f724de2fd493a602fe99bef392fdba32b06d66/source_subfolder/configure --enable-static --disable-shared --prefix=C:/Users/me/.conan/data/libiconv/1.17/_/_/package/11f724de2fd493a602fe99bef392fdba32b06d66
libiconv/1.17:
libiconv/1.17: ERROR: Package '11f724de2fd493a602fe99bef392fdba32b06d66' build failed
libiconv/1.17: WARN: Build folder C:\Users\me\.conan\data\libiconv\1.17\_\_\build\11f724de2fd493a602fe99bef392fdba32b06d66
ERROR: libiconv/1.17: Error in build() method, line 140
        autotools = self._configure_autotools()
while calling '_configure_autotools', line 126
        autotools.configure(args=configure_args, host=host, build=build)
        ConanException: Cannot recognize the Windows subsystem, install MSYS2/cygwin or specify a build_require to apply it.
```
</details>

**Comment by uilianries** [comment]
@TheClonerx please, consider using os.subsystem in your profile: https://docs.conan.io/en/latest/systems_cross_building/windows_subsystems.html#windows-subsystems

**Comment by TheClonerx** [comment]
@uilianries 
> If you are building specific software for the subsystem, you have to specify a value for the setting `os.subsystem`, if you are only using the subsystem for taking benefit of the UNIX tools but generating native Windows software, you shouldn’t specify it.

I'm not building this in order to run it in a subsystem, i just want to be able to build it without Visual Studio and the other Microsoft provided tools

**Comment by uilianries** [comment]
@TheClonerx YOu need subsystem because you are using conf. It will be the only way on Conan 2.0

**Comment by TheClonerx** [comment]
So it won't be possible to build with unix tools without targeting a subsystem?

**Comment by uilianries** [comment]
Yes, it's possible, just read some topics below, in the link that I sent. There is a profile using gcc on Windows and mingw or msys as build requirement.

**Comment by kambala-decapitator** [comment]
> Yes, it's possible, just read some topics below, in the link that I sent. There is a profile using gcc on Windows and mingw or msys as build requirement.

I'll be honest: that page is quite cryptic, especially to someone who isn't dealing with msvc/mignw every day.

On the page I'm reading:

> If you are building specific software for the subsystem, you have to specify a value for the setting os.subsystem, if you are only using the subsystem for taking benefit of the UNIX tools but generating native Windows software, you shouldn’t specify it.

@uilianries and you're saying:

> consider using os.subsystem in your profile
> YOu need subsystem because you are using conf. It will be the only way on Conan 2.0

is the page simply outdated? And `os.subsystem` **must** be specified in Windows profiles? Or are you actually referring to `tools.microsoft.bash:subsystem`?

I'm still struggling with https://github.com/conan-io/conan/issues/12323 and can't understand where it's an issue with my local environment or a bug somewhere in Conan. Any help appreciated.

**Comment by uilianries** [comment]
@kambala-decapitator It's not outdated, you are getting mixed by Conan 1.x with 2.x features.

`global.conf` is the feature to be used with Conan 2.0. Meanwhile, recipes in CCI are adopting it, like `msys2` package. More info: https://docs.conan.io/en/latest/migrating_to_2.0/recipes.html#windows-subsystems

The point is, if you are consuming `msys2` Conan package, you should configure according the link above, using `global.conf` and any environment variable will be ignored.

Packages which are not ported, they still use [CONAN_BASH_PATH](https://docs.conan.io/en/latest/reference/env_vars.html#conan-bash-path).

How to know it which one should be used? Only checking by recipe. But you can add environment variables to your Conan profile, include your bash path: https://docs.conan.io/en/latest/reference/profiles.html

The reason is because was too fragile running gcc on windows due those many subsystems, so Conan 2.x adapted it. Conan 1.x can't use global.conf default, because the backward compatibility should be respected.

**Comment by kambala-decapitator** [comment]
>  if you are consuming msys2 Conan package, you should configure according the link above, using global.conf and any environment variable will be ignored.
> Packages which are not ported, they still use [CONAN_BASH_PATH](https://docs.conan.io/en/latest/reference/env_vars.html#conan-bash-path).

@uilianries I'm not consuming the package because it is available only for x64. But I have configured both `bash_path` in `conan.conf` (which is analogous to `CONAN_BASH_PATH` if I read it right) and `tools.microsoft.bash:path` / `tools.microsoft.bash:subsystem=msys2`.

Still Autotools-based recipes don't build and I don't understand how to figure out what's wrong and on which side. Enabling Conan's debug output didn't shed any light on me.
